ads/google/utils.js

```diff
@@ -25,52 +25,69 @@ import {user} from '../../src/log';
  *   multi-size dimensions.
  * @param {number} primaryWidth The primary width of the ad slot.
  * @param {number} primaryHeight The primary height of the ad slot.
- * @param {!Array<Array<number>>=} opt_dimensions An array into which to put
- *   the multi-size dimensions.
  * @param {boolean} multiSizeValidation A flag that if set to true will enforce
  *   the rule that ensures multi-size dimensions are no less than 2/3rds of
  *   their primary dimension's counterpart.
- * @return {!Array<Array<number>>} An array of dimensions.
+ * @param {boolean} strict If set to true, this indicates that a single
+ *   malformed size should cause the entire multi-size data string to be
+ *   abnadoned. If set to false, then malformed sizes will be ignored, and the
+ *   remainder of the string will be parsed for any additional sizes.
+ *   Additionally, errors will only be reported if this flag is set to true.
+ * @return {?Array<Array<number>>} An array of dimensions.
  */
 export function getMultiSizeDimensions(
     multiSizeDataStr,
     primaryWidth,
     primaryHeight,
     multiSizeValidation,
-    opt_dimensions) {
+    strict) {
 
-  const dimensions = opt_dimensions || [];
+  const dimensions = [];
   const arrayOfSizeStrs = multiSizeDataStr.split(',');
 
-  arrayOfSizeStrs.forEach(sizeStr => {
+  for (let i = 0; i < arrayOfSizeStrs.length; i++) {
 
+    const sizeStr = arrayOfSizeStrs[i];
     const size = sizeStr.split('x');
 
-    // Make sure that each size is specified in the form val1xval2.
+    // Make sure that each size is specified in the form WxH.
     if (size.length != 2) {
       user().error('AMP-AD', `Invalid multi-size data format '${sizeStr}'.`);
-      return;
+      if (strict) {
+        return null;
+      }
+      continue;
     }
 
     const width = Number(size[0]);
     const height = Number(size[1]);
 
-    // Make sure that both dimensions given are numbers.
+    // Make sure that both dimensions given are positive numbers.
     if (!validateDimensions(width, height,
-        w => isNaN(w),
-        h => isNaN(h),
-        ({badDim, badVal}) =>
-          `Invalid ${badDim} of ${badVal} given for secondary size.`)) {
-      return;
+        w => isNaN(w) || w <= 0,
+        h => isNaN(h) || h <= 0,
+        badParams => badParams.map(badParam =>
```

Comment:
Seems like you do a similar operation multiple times where you do a badParams mapping and then return or continue based on strict.  Consider creating a helper function to save cost & paste